src/main/java/org/snomed/snowstorm/rest/ConceptController.java

```diff
@@ -580,6 +580,40 @@ public ExpressionStringPojo getConceptNormalForm(
 
 		return new ExpressionStringPojo(expression.toString(includeTerms));
 	}
+	
+	@PostMapping(value = "/{branch}/concepts/donate")
+	@PreAuthorize("hasPermission('AUTHOR', #branch)")
+	@JsonView(value = View.Component.class)
+	public Collection<ConceptMini> donateConcepts(
+			@ApiParam(value = "Destination branch where the concepts are copied to.")
+			@PathVariable String branch,
+			@ApiParam(value = "Source branch where the concepts are selected from.")
+			@RequestParam String sourceBranch,
+			@ApiParam(value = "ECL expression for selecting concepts to copy between the branches.")
+			@RequestParam String ecl,
+			@ApiParam(value = "Include dependant components. Defaults to false.")
+			@RequestParam(required = false, defaultValue = "false") boolean includeDependencies) throws ServiceException {
+		String sourceBranchPath = BranchPathUriUtil.decodePath(sourceBranch);
+		String destinationBranchPath = BranchPathUriUtil.decodePath(branch);
+
+		final List<LanguageDialect> languageDialects = List.of(new LanguageDialect(Config.DEFAULT_LANGUAGE_CODE, Long.parseLong(Concepts.US_EN_LANG_REFSET)));
+
+		// Get concepts
+		QueryService.ConceptQueryBuilder queryBuilder = queryService.createQueryBuilder(true)
+				.activeFilter(true)
+				.ecl(ecl)
+				.isReleased(true)
+				.resultLanguageDialects(languageDialects);
+
+		List<ConceptMini> conceptsFound = queryService.search(queryBuilder, sourceBranchPath, LARGE_PAGE).getContent();
+		List<ConceptMini> conceptsDonated = new ArrayList<>();
+
+		for (ConceptMini concept : conceptsFound) {
```

Comment:
@astro-snail The ECL results potentially will return concepts are already in the core module. We should add logic here to prevent these being selected again.